modules/data.atmosphere/R/download.NEONmet.R

```diff
@@ -0,0 +1,486 @@
+          csvVar <- csvData$tempSingleMean
+          csvQF <- csvData$finalQF
+          csvVar[which(csvQF!=0)]=NA
+          ncdata[arrLoc] <- udunits2::ud.convert(csvVar,"celsius", "K")
+        }
+      }
+    }
+    ncdf4::ncvar_put(nc, varid = airT.var, vals = ncdata)
+
+    # NEON.DP1.00004 Pressure
+    pSurfLoc = grep("DP1.00004",availProducts)
+    if (length(pSurfLoc)!=0) { 
+      pSurfDates = unlist(availDates[pSurfLoc[1]])
+      pSurfGoodDates = which((pSurfDates >= startMon) & (pSurfDates <= endMon))
+      npSurf = length(pSurfGoodDates)
+    } else {npSurf=0}
+    if (npSurf>0) {
+      Psurf.var <- ncdf4::ncvar_def(name = "air_pressure", units = "Pa", dim = xytdim)
+      nc <- ncdf4::ncvar_add(nc = nc, v = Psurf.var, verbose = verbose)
+      ncdata <- rep(FillValue,length(datetime))
+      for (mon in pSurfDates[pSurfGoodDates]) {
+          neonData <- nneo::nneo_data(product_code = availProducts[pSurfLoc[1]], site_code = site, year_month = mon)
+          if (length(neonData$data$urls)>0) {
+            #Extract and read 30 minute data from the highest vertical level among files returned
+            url30 <- tail(sort(neonData$data$urls[grep("*.00000.000.*_30min",neonData$data$urls)]),1)
+            csvData <- read.csv(url30,stringsAsFactors=FALSE,header=TRUE)
+            #Retreive time dimension and figure out where in array to put it
+            csvDateTime <- as.POSIXct(gsub("T"," ",csvData$startDateTime),tz="UTC")
+            arrLoc <- floor(as.numeric(difftime(csvDateTime,datetime[1],tz="UTC",units="hours"))*2)+1
+            csvVar <- csvData$staPresMean
+            csvQF <- csvData$staPresFinalQF
+            csvVar[which(csvQF!=0)]=NA
+            ncdata[arrLoc] <- udunits2::ud.convert(csvVar,"kPa", "Pa")
+          }
+      }
+      ncdf4::ncvar_put(nc, varid = Psurf.var, vals = ncdata)
+    }
+      
+    # NEON.DP1.00024 PAR
+    PARLoc = grep("DP1.00024",availProducts)
+    if (length(PARLoc)!=0) { 
+      PARDates = unlist(availDates[PARLoc[1]])
+      PARGoodDates = which((PARDates >= startMon) & (PARDates <= endMon))
+      nPAR = length(PARGoodDates)
+    } else {nPAR=0}
+    if (nPAR>0) {
+      PAR.var <- ncdf4::ncvar_def(name = "surface_downwelling_photosynthetic_photon_flux_in_air", 
+                                  units = "mol m-2 s-1", 
+                                  dim = xytdim)
+      nc <- ncdf4::ncvar_add(nc = nc, v = PAR.var, verbose = verbose)
+      ncdata <- rep(FillValue,length(datetime))
+      for (mon in PARDates[PARGoodDates]) {
+        neonData <- nneo::nneo_data(product_code = availProducts[PARLoc[1]], site_code = site, year_month = mon)
+        if (length(neonData$data$urls)>0) {
+          #Extract and read 30 minute data from the highest vertical level among files returned
+          url30 <- tail(sort(neonData$data$urls[grep("*.00000.000.*_30min",neonData$data$urls)]),1)
```

Comment:
Also, it seems like there is a lot of repetition in here. I would suggest trying to write a generic function that gets called multiple times rather than repeating code with minor changes.